Applications/Calculator/CalculatorWidget.cpp

```diff
@@ -0,0 +1,204 @@
+    button_toggle_sign.set_text("+/-");
+    add_button(button_toggle_sign, Calculator::Operation::ToggleSign);
+
+    auto& button_add = *new GButton(this);
+    button_add.move_to(172, 177);
+    button_add.set_foreground_color(Color::NamedColor::Red);
+    button_add.set_text("+");
+    add_button(button_add, Calculator::Operation::Add);
+
+    auto& button_subtract = *new GButton(this);
+    button_subtract.move_to(172, 144);
+    button_subtract.set_foreground_color(Color::NamedColor::Red);
+    button_subtract.set_text("-");
+    add_button(button_subtract, Calculator::Operation::Subtract);
+
+    auto& button_multiply = *new GButton(this);
+    button_multiply.move_to(172, 111);
+    button_multiply.set_foreground_color(Color::NamedColor::Red);
+    button_multiply.set_text("*");
+    add_button(button_multiply, Calculator::Operation::Multiply);
+
+    auto& button_divide = *new GButton(this);
+    button_divide.move_to(172, 78);
+    button_divide.set_foreground_color(Color::NamedColor::Red);
+    button_divide.set_text("/");
+    add_button(button_divide, Calculator::Operation::Divide);
+
+    auto& button_sqrt = *new GButton(this);
+    button_sqrt.move_to(211, 78);
+    button_sqrt.set_foreground_color(Color::NamedColor::Blue);
+    button_sqrt.set_text("sqrt");
+    add_button(button_sqrt, Calculator::Operation::Sqrt);
+
+    auto& button_inverse = *new GButton(this);
+    button_inverse.move_to(211, 144);
+    button_inverse.set_foreground_color(Color::NamedColor::Blue);
+    button_inverse.set_text("1/x");
+    add_button(button_inverse, Calculator::Operation::Inverse);
+
+    auto& button_percent = *new GButton(this);
+    button_percent.move_to(211, 111);
+    button_percent.set_foreground_color(Color::NamedColor::Blue);
+    button_percent.set_text("%");
+    add_button(button_percent, Calculator::Operation::Percent);
+
+    auto& button_equals = *new GButton(this);
+    button_equals.move_to(211, 177);
+    button_equals.set_foreground_color(Color::NamedColor::Red);
+    button_equals.set_text("=");
+    button_equals.on_click = [this](GButton&) {
+        double argument = m_keypad.value();
+        double res = m_calculator.finish_operation(argument);
+        m_keypad.set_value(res);
+        update_display();
+    };
+    add_button(button_equals);
+}
+
+CalculatorWidget::~CalculatorWidget()
+{
+    delete m_entry;
+    delete m_label;
+    for (GButton* button : m_buttons)
+        delete button;
+}
+
+void CalculatorWidget::add_button(GButton& button, Calculator::Operation operation)
+{
+    add_button(button);
+    button.on_click = [this, operation](GButton&) {
+        double argument = m_keypad.value();
+        double res = m_calculator.begin_operation(operation, argument);
+        m_keypad.set_value(res);
+        update_display();
+    };
+}
+
+void CalculatorWidget::add_button(GButton& button, int digit)
+{
+    add_button(button);
+    button.set_text(String::format("%d", digit));
```

Comment:
There's a helper for this to avoid having to use `String::format` directly:

```suggestion
    button.set_text(String::number(digit));
```